aiohttp/connector.py

```diff
@@ -390,6 +390,13 @@ def connect(self, req):
                 if self._closed:
                     proto.close()
                     raise ClientConnectionError("Connector is closed.")
+            except:
+                # signal to waiter
+                waiters = self._waiters[key]
+                if waiters:
+                    waiter = waiters.pop(0)
```

Comment:
I don't think this is required, this could cause IndexError on line 379

Reply:
maybe you should just call `_release_waiter`?

Reply:
Yep, figured it out after writing tests.
Please wait for update -- polishing test suite